extensions/pkg/controller/infrastructure/reconciler.go

```diff
@@ -218,3 +236,18 @@ func (r *reconciler) removeAnnotation(ctx context.Context, logger logr.Logger, i
 	}
 	return nil
 }
+
+func (r *reconciler) validateConfig(ctx context.Context, infrastructure *extensionsv1alpha1.Infrastructure) error {
+	if r.configValidator == nil {
+		return nil
+	}
+
+	if allErrs := r.configValidator.Validate(ctx, infrastructure); len(allErrs) > 0 {
+		if internalErrs := allErrs.Filter(field.NewErrorTypeMatcher(field.ErrorTypeInternal)); len(internalErrs) > 0 {
+			return internalErrs.ToAggregate()
+		}
+		return gardencorev1beta1helper.NewErrorWithCodes(allErrs.ToAggregate().Error(), gardencorev1beta1.ErrorConfigurationProblem)
```

Comment:
As described above, this will fail with `InternalError` and the error returned will not be reported as `ErrorConfigurationProblem` here, instead it will be classified as whatever error it is by the existing (regex-based) classification mechanism. I think it would be far too complex for `ConfigValidators` to try to detect errors such as `InsufficientPrivileges` or `Unauthorized` instead of just failing with `InternalError`.